doc/migration/version-upgrade-guide.md

```diff
@@ -416,18 +416,22 @@ Upon updating the project to `v0.8.2` the following breaking changes apply:
     ```
     // Pinned to kubernetes-1.14.1
     replace (
-    	k8s.io/api => k8s.io/api kubernetes-1.14.1
-    	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver kubernetes-1.14.1
-    	k8s.io/apimachinery => k8s.io/apimachinery kubernetes-1.14.1
-    	k8s.io/client-go => k8s.io/client-go kubernetes-1.14.1
-    	k8s.io/cloud-provider => k8s.io/cloud-provider kubernetes-1.14.1
+    	k8s.io/api => k8s.io/api v0.0.0-20190409021203-6e4e0e4f393b
+    	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20190409022649-727a075fdec8
+    	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20190404173353-6a84e37a896d
+    	k8s.io/client-go => k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
+    	k8s.io/cloud-provider => k8s.io/cloud-provider v0.0.0-20190409023720-1bc0c81fa51d
```

Comment:
I don't think these changes are necessary, since Go does the lookup from the `kubernetes-1.14.1` branch to the correct psuedoversion string automatically.

Reply:
Hi Joe - I'll be the first to admit that I'm really green here.  When I go back to the original go.mod, these are the errors that I get:
```
GO111MODULE=on operator-sdk generate k8s
FATA[0000] Parse go.mod: go.mod:43: invalid module version k8s.io/api: version must be of the form v1.2.3
go.mod:44: invalid module version k8s.io/apiextensions-apiserver: version must be of the form v1.2.3
go.mod:45: invalid module version k8s.io/apimachinery: version must be of the form v1.2.3
go.mod:46: invalid module version k8s.io/client-go: version must be of the form v1.2.3
go.mod:47: invalid module version k8s.io/cloud-provider: version must be of the form v1.2.3
go.mod:54: invalid module version github.com/prometheus/prometheus: version must be of the form v1.2.3 
```
Is there maybe some other way to tell go to use a branch?  I've tried:
```
	k8s.io/api => k8s.io/api kubernetes-1.14.1
```
and
```
	k8s.io/api => k8s.io/api v0.0.0-kubernetes-1.14.1
```

Reply:
I'm guessing `operator-sdk generate k8s` doesn't invoke the pieces of the Go toolchain that does the lookup. Can you try `GO111MODULE=on go mod tidy` before running the `generate` subcommand?

Reply:
ok, I see.  `go mod tidy` replaced the `kubernetes-1.14.1` branch with the actual commit that it represents, in go.mod.  So this is just user error on my part.  I was going to question why the `go.mod` in my new operator-sdk project doesn't look like this, but it must be that I ran `go mod tidy` on that project before I checked what was in `go.mod` and so I saw the end result instead of the branch name.  Thanks for guiding me thru this.

Reply:
Actually, I the `operator-sdk new` command runs `go build ./...` under the hood to validate the newly scaffolded project. `go build` is one of the Go toolchain pieces that does the go.mod dependency lookup and resolution, so that's probably why you never saw the branch names, tags, or commit hashes in your project when you originally created it.